DatabaseException: Error updating 'CVE-2020-36569'

**Describe the bug**
     classpath 'com.android.tools.build:gradle:4.1.1'
gradle version is: 7.0.2

after run ./gradlew dependencyCheckAnalyze
in logs falls down, this error:

```> Task :dependencyCheckUpdate
org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569'
org.owasp.dependencycheck.data.update.exception.UpdateException: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569'
	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.processFiles(ProcessTask.java:157)
	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.call(ProcessTask.java:114)
	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.call(ProcessTask.java:41)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569'
	at org.owasp.dependencycheck.data.nvdcve.CveDB.updateVulnerability(CveDB.java:823)
	at org.owasp.dependencycheck.data.update.nvd.NvdCveParser.parse(NvdCveParser.java:114)
Caused by: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569'

	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.importJSON(ProcessTask.java:141)
	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.processFiles(ProcessTask.java:154)
	... 6 more
Caused by: org.h2.jdbc.JdbcBatchUpdateException: Value too long for column "VERSIONENDEXCLUDING CHARACTER VARYING(60)": "'0.0.0-20160722212129-ac0cc4484ad4_before_v0.0.0-20200131131040-063a3fb69896' (75)"; SQL statement:
INSERT INTO software (cveid, cpeEntryId, versionEndExcluding, versionEndIncluding, versionStartExcluding, versionStartIncluding, vulnerable) VALUES (?, ?, ?, ?, ?, ?, ?) [22001-214]
	at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1269)
	at org.apache.commons.dbcp2.DelegatingStatement.executeBatch(DelegatingStatement.java:241)
	at org.apache.commons.dbcp2.DelegatingStatement.executeBatch(DelegatingStatement.java:241)
	at org.owasp.dependencycheck.data.nvdcve.CveDB.executeBatch(CveDB.java:1248)
	at org.owasp.dependencycheck.data.nvdcve.CveDB.updateVulnerabilityInsertSoftware(CveDB.java:1098)
Caused by: org.h2.jdbc.JdbcBatchUpdateException: Value too long for column "VERSIONENDEXCLUDING CHARACTER VARYING(60)": "'0.0.0-20160722212129-ac0cc4484ad4_before_v0.0.0-20200131131040-063a3fb69896' (75)"; SQL statement:

	at org.owasp.dependencycheck.data.nvdcve.CveDB.updateVulnerability(CveDB.java:816)
	... 9 more
```

**Version of dependency-check used**
classpath 'org.owasp:dependency-check-gradle:7.2.1'
classpath 'org.owasp:dependency-check-gradle:7.4.3'

**Log file**
When reporting errors, 99% of the time log file output is required. Please post the log file as a [gist](https://gist.github.com/) and provide a link in the new issue.

**To Reproduce**
Steps to reproduce the behavior:
1. run ./gradlew dependencyCheckAnalyze
2. See error

**Expected behavior**
task started and scan completed

**Additional context**
Add any other context about the problem here.

Same problem here

> `[ERROR] Unable to continue dependency-check analysis.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:55 min
[INFO] Finished at: 2023-01-06T10:36:47Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:7.4.3:check (default) on project config-service: Fatal exception(s) analyzing config-service: One or more exceptions occurred during analysis:
[ERROR] 	UpdateException: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569'
[ERROR] 		caused by DatabaseException: Error updating 'CVE-2020-36569'
[ERROR] 		caused by JdbcBatchUpdateException: Value too long for column "VERSIONENDEXCLUDING CHARACTER VARYING(60)": "'0.0.0-20160722212129-ac0cc4484ad4_before_v0.0.0-20200131131040-063a3fb69896' (75)"; SQL statement:
[ERROR] INSERT INTO software (cveid, cpeEntryId, versionEndExcluding, versionEndIncluding, versionStartExcluding, versionStartIncluding, vulnerable) VALUES (?, ?, ?, ?, ?, ?, ?) [22001-214]
[ERROR] 	NoDataException: No documents exist`

builds are failing due to [ERROR] org.owasp.dependencycheck.data.nvdcve.DatabaseException: Unable to retrieve id for new vulnerability for 'CVE-2022-32923'
org.owasp.dependencycheck.data.update.exception.UpdateException: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Unable to retrieve id for new vulnerability for 'CVE-2022-32923'

> > Any idea when its going to hit the Maven Repo https://mvnrepository.com/artifact/org.owasp/dependency-check-maven
> 
> It's already available: https://repo1.maven.org/maven2/org/owasp/dependency-check-maven/7.4.4/ mvnrepository.com and search.maven.org don't index new artifacts immediately, but that doesn't mean the new version isn't ready to be downloaded.

Seems to be failing still when we run this command org.owasp:dependency-check-maven:RELEASE:check 

[2023-01-06T19:25:47.041Z] [2023-01-06T19:25:47.040Z] [ERROR] org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569'

[2023-01-06T19:25:47.041Z] [2023-01-06T19:25:47.041Z] org.owasp.dependencycheck.data.update.exception.UpdateException: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569'

After updating from 7.4.3 to 7.4.4, I have this error in my Gitlab CI:

```
Caused by: org.h2.jdbc.JdbcSQLTransientException: Row not found when trying to delete from index "PUBLIC.IDXSOFTWARECPE: 12109601" [90112-214]

[ERROR] Failed to execute goal org.owasp:dependency-check-maven:7.4.4:aggregate (default) on project my-project: Fatal exception(s) analyzing my-project: One or more exceptions occurred during analysis:
[ERROR] 	UpdateException: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Unable to retrieve id for new vulnerability for 'CVE-2020-25021'
[ERROR] 		caused by DatabaseException: Unable to retrieve id for new vulnerability for 'CVE-2020-25021'
[ERROR] 		caused by JdbcSQLTimeoutException: Timeout trying to lock table {0}; SQL statement:
[ERROR] DELETE FROM software WHERE cveid = ? [50200-214]
[ERROR] 		caused by DbException: Row not found when trying to delete from index "PUBLIC.IDXSOFTWARECPE: 12109601" [90112-214]
[ERROR] 		caused by JdbcSQLTransientException: Row not found when trying to delete from index "PUBLIC.IDXSOFTWARECPE: 12109601" [90112-214]
[ERROR] 	NoDataException: No documents exist
[ERROR] -> [Help 1]
```

¿What can I do? Thanks in advance

Hi All. Thanks for all the work you're all putting in.
I'm working for NHS (UK National Health Service) and unfortunately all our pipelines are still broken. We use Maven and we're still seeing the following...

`[ERROR] Failed to execute goal org.owasp:dependency-check-maven:7.1.0:aggregate (default-cli) on project bcss-regression: Fatal exception(s) analyzing bcss-regression: One or more exceptions occurred during analysis:
[ERROR] 	UpdateException: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569'
[ERROR] 		caused by DatabaseException: Error updating 'CVE-2020-36569'
[ERROR] 		caused by JdbcBatchUpdateException: Value too long for column "VERSIONENDEXCLUDING CHARACTER VARYING(60)": "'0.0.0-20160722212129-ac0cc4484ad4_before_v0.0.0-20200131131040-063a3fb69896' (75)"; SQL statement:
[ERROR] INSERT INTO software (cveid, cpeEntryId, versionEndExcluding, versionEndIncluding, versionStartExcluding, versionStartIncluding, vulnerable) VALUES (?, ?, ?, ?, ?, ?, ?) [22001-210]`

I'm about to back out all dependency scanning, but before I do, has anyone any suggestions? We use docker containers so I thought it may be docker caching, but I've clearer the cache.
Thanks

> Hi All. Thanks for all the work you're all putting in. I'm working for NHS (UK National Health Service) and unfortunately all our pipelines are still broken. We use Maven and we're still seeing the following...
> 
> `[ERROR] Failed to execute goal org.owasp:dependency-check-maven:7.1.0:aggregate (default-cli) on project bcss-regression: Fatal exception(s) analyzing bcss-regression: One or more exceptions occurred during analysis: [ERROR] UpdateException: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Error updating 'CVE-2020-36569' [ERROR] caused by DatabaseException: Error updating 'CVE-2020-36569' [ERROR] caused by JdbcBatchUpdateException: Value too long for column "VERSIONENDEXCLUDING CHARACTER VARYING(60)": "'0.0.0-20160722212129-ac0cc4484ad4_before_v0.0.0-20200131131040-063a3fb69896' (75)"; SQL statement: [ERROR] INSERT INTO software (cveid, cpeEntryId, versionEndExcluding, versionEndIncluding, versionStartExcluding, versionStartIncluding, vulnerable) VALUES (?, ?, ?, ?, ?, ?, ?) [22001-210]`
> 
> I'm about to back out all dependency scanning, but before I do, has anyone any suggestions? We use docker containers so I thought it may be docker caching, but I've clearer the cache. Thanks

you must be upgrade to `7.4.4` version. it will fix the problem.
thanks the all contributor of this project.

--- a/core/src/main/resources/data/initialize_mssql.sql
+++ b/core/src/main/resources/data/initialize_mssql.sql
@@ -45,8 +45,8 @@ CREATE TABLE cpeEntry (id INT identity(1,1) PRIMARY KEY, part CHAR(1), vendor VA
     version VARCHAR(255), update_version VARCHAR(255), edition VARCHAR(255), lang VARCHAR(20), sw_edition VARCHAR(255), 
     target_sw VARCHAR(255), target_hw VARCHAR(255), other VARCHAR(255), ecosystem VARCHAR(255));
 
-CREATE TABLE software (cveid INT, cpeEntryId INT, versionEndExcluding VARCHAR(60), versionEndIncluding VARCHAR(60), 
-                       versionStartExcluding VARCHAR(60), versionStartIncluding VARCHAR(60), vulnerable BIT
+CREATE TABLE software (cveid INT, cpeEntryId INT, versionEndExcluding VARCHAR(100), versionEndIncluding VARCHAR(100), 
+                       versionStartExcluding VARCHAR(100), versionStartIncluding VARCHAR(100), vulnerable BIT
     , CONSTRAINT FK_SoftwareCve FOREIGN KEY (cveid) REFERENCES vulnerability(id) ON DELETE CASCADE
     , CONSTRAINT FK_SoftwareCpeProduct FOREIGN KEY (cpeEntryId) REFERENCES cpeEntry(id));
 

